### PR TITLE
CI Infra for CPU Support

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*
Building CI infrastructure for CPU support.
Uses the same infrastructure as done by lint check.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflow_scripts/e2e_cpu_check.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Move to parent directory
+cd ../../
+
+set -ex
+
+sh ./tests/end2end-tests/setup.sh
+sh ./tests/end2end-tests/create_data.sh
+sh ./tests/end2end-tests/tools/test_mem_est.sh
```
Comment (classicsong) [inline .github/workflow_scripts/e2e_cpu_check.sh:8]: This is not needed.

**Review by classicsong — APPROVED** [review]
Overall LGTM

File: .github/workflow_scripts/e2e_cpu_check.sh (hunk 1 above)
Comment (prateekdesai04) [reply .github/workflow_scripts/e2e_cpu_check.sh:8]: we would need _create_data.sh_ for partitioning

Comment (classicsong) [reply .github/workflow_scripts/e2e_cpu_check.sh:8]: I mean we do not need `sh ./tests/end2end-tests/tools/test_mem_est.sh`

Comment (prateekdesai04) [reply .github/workflow_scripts/e2e_cpu_check.sh]: Done

**Pull request merged**